src/Microsoft.Framework.Runtime/ApplicationHostContext.cs

```diff
@@ -173,5 +176,35 @@ public IServiceProvider ServiceProvider
         public string ProjectDirectory { get; private set; }
 
         public string PackagesDirectory { get; private set; }
+
+
+
+        public IEnumerable<ICompilationMessage> LockFileDiagnostics
```

Comment:
Because sometimes we only need diagnostics for lock file. Check the case below:
![capture](https://cloud.githubusercontent.com/assets/1383883/8737233/d97462d6-2bd4-11e5-97e4-bf661dec58d1.PNG)

In this case, we should only include lock file diagnostics to the error message, instead of all diagnostics. Otherwise, you'll see "unable to resolve Newtonsoft.Json" twice